internal/fileutil/fileutil.go

```diff
@@ -55,3 +55,10 @@ func FileContains(path string, substring string) (bool, error) {
 	}
 	return strings.Contains(string(data), substring), nil
 }
+
+// ParseVersionedPackage checks if the given package is a versioned package (`python@3.10`)
+// and returns its name and version
+func ParseVersionedPackage(pkg string) (string, string, bool) {
```

Comment:
fileutil doesn't seem like the right place for this. Maybe `input.go` in nix package? (I think this creates a dependency cycle though) Or we can create a new `devpkg` (I've been meaning to move `nix.Input` to own package, but that's out of scope for this PR)

Reply:
That's right. I'll create a new `devpkg` first and then move `nix.Input` to that package and rename that package.

Reply:
I'll learn more about input and then think about adding a package for nix packages.